### PR TITLE
Create FastBridgeCompatibleBtcWallet class in order to support fast BTC feature

### PR DESCRIPTION
**Pull request by guidohernan93**
- Refactor RegisterBtcWallet class
- Refactor BridgeUtils methods involving spendWallet and noSpendWallet as boolean parameter is added to check if wallet is compatible with fast bridge
- Refactored all the tests that made use of these methods and add test for FastBridgeCompatibleBtcWallet class

### CODE REVIEW

File: rskj-core/src/main/java/co/rsk/peg/FastBridgeCompatibleBtcWallet.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    @Nullable
+    @Override
+    public RedeemData findRedeemDataFromScriptHash(byte[] payToScriptHash) {
+        Optional<FastBridgeFederationInformation> fastBridgeFederationInformation =
+            this.bridgeStorageProvider.getFastBridgeFederationInformation(payToScriptHash);
+
+        if (fastBridgeFederationInformation.isPresent()) {
+            Optional<Federation> destinationFederation = getDestinationFederation(payToScriptHash);
```
Comment (josedahlquist) [inline rskj-core/src/main/java/co/rsk/peg/FastBridgeCompatibleBtcWallet.java:28]: You should be using the `fastBridgeFederationInformation` payToScriptHash.
The way you are trying to get the destination address won't work

**Review by josedahlquist — CHANGES_REQUESTED** [review]

File: rskj-core/src/main/java/co/rsk/peg/FastBridgeCompatibleBtcWallet.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    @Nullable
+    @Override
+    public RedeemData findRedeemDataFromScriptHash(byte[] payToScriptHash) {
+        Optional<FastBridgeFederationInformation> fastBridgeFederationInformation =
+            this.bridgeStorageProvider.getFastBridgeFederationInformation(payToScriptHash);
+
+        if (fastBridgeFederationInformation.isPresent()) {
+            Optional<Federation> destinationFederation = getDestinationFederation(
+                fastBridgeFederationInformation.get().getFederationScriptHash());
```
Comment (josedahlquist) [inline rskj-core/src/main/java/co/rsk/peg/FastBridgeCompatibleBtcWallet.java:29]:
```suggestion
            FastBridgeFederationInformation fastBridgeFederationInformationInstance = fastBridgeFederationInformation.get(); 
            Optional<Federation> destinationFederation = getDestinationFederation(
                fastBridgeFederationInformationInstance.getFederationScriptHash()
            );
```
Seems like you are "getting" the optional many times, maybe get it just once into a variable

File: rskj-core/src/test/java/co/rsk/peg/FastBridgeCompatibleBtcWalletTest.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        Script p2SHOutputScript = ScriptBuilder.createP2SHOutputScript(fastBridgeRedeemScript);
+        byte[] fastBridgeFederationP2SH = p2SHOutputScript.getPubKeyHash();
+
+        FastBridgeFederationInformation fastBridgeFederationInformation =
+            new FastBridgeFederationInformation(derivationArgumentsHash,
+                federation.getP2SHScript().getPubKeyHash());
+
+        when(provider.getFastBridgeFederationInformation(fastBridgeFederationP2SH)).thenReturn
+            (Optional.of(fastBridgeFederationInformation)
+        );
```
Comment (josedahlquist) [inline rskj-core/src/test/java/co/rsk/peg/FastBridgeCompatibleBtcWalletTest.java:64]:
```suggestion
        when(provider.getFastBridgeFederationInformation(fastBridgeFederationP2SH))
           .thenReturn(Optional.of(fastBridgeFederationInformation));
```

**Review by josedahlquist — APPROVED** [review]

File: rskj-core/src/main/java/co/rsk/peg/BridgeUtils.java [hunk 4]
```diff
@@ -49,22 +49,37 @@
...
+
         federations.forEach(federation -> wallet.addWatchedAddress(federation.getAddress(), federation.getCreationTime().toEpochMilli()));
         return wallet;
     }
 
-    public static Wallet getFederationSpendWallet(Context btcContext, Federation federation, List<UTXO> utxos) {
-        return getFederationsSpendWallet(btcContext, Collections.singletonList(federation), utxos);
+    public static Wallet getFederationSpendWallet(Context btcContext, Federation federation, List<UTXO> utxos,
+        boolean isFastBridgeCompatible, BridgeStorageProvider storageProvider) {
+        return getFederationsSpendWallet(btcContext, Collections.singletonList(federation), utxos,
```
Comment (pamgonzalez) [inline rskj-core/src/main/java/co/rsk/peg/BridgeUtils.java:71]: please, try to maintain the code standards.

File: rskj-core/src/main/java/co/rsk/peg/BridgeUtils.java [hunk 5]
```diff
@@ -49,22 +49,37 @@
...
+            isFastBridgeCompatible, storageProvider);
     }
 
-    public static Wallet getFederationsSpendWallet(Context btcContext, List<Federation> federations, List<UTXO> utxos) {
-        Wallet wallet = new BridgeBtcWallet(btcContext, federations);
+    public static Wallet getFederationsSpendWallet(Context btcContext, List<Federation> federations,
+        List<UTXO> utxos, boolean isFastBridgeCompatible, BridgeStorageProvider storageProvider) {
```
Comment (pamgonzalez) [inline rskj-core/src/main/java/co/rsk/peg/BridgeUtils.java:76]: please, try to maintain the code standards.

File: rskj-core/src/main/java/co/rsk/peg/FastBridgeCompatibleBtcWallet.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+import co.rsk.bitcoinj.wallet.RedeemData;
+import co.rsk.peg.fastbridge.FastBridgeFederationInformation;
+import java.util.List;
+import java.util.Optional;
+import javax.annotation.Nullable;
+
+public class FastBridgeCompatibleBtcWallet extends BridgeBtcWallet {
+    private final BridgeStorageProvider bridgeStorageProvider;
+
+    public FastBridgeCompatibleBtcWallet(Context btcContext, List<Federation> federations,
```
Comment (pamgonzalez) [inline rskj-core/src/main/java/co/rsk/peg/FastBridgeCompatibleBtcWallet.java:15]: please, try to maintain the code standards.

**Review by josedahlquist — APPROVED** [review]

**Review by pamgonzalez — APPROVED** [review]

**Pull request merged**